examples/aa-simple-dapp/src/hooks/usePlugin.ts

```diff
@@ -0,0 +1,56 @@
+import {
+  UninstallPluginParams,
+  accountLoupeDecorators,
+  pluginManagerDecorator,
+  type IMSCA,
+  type Plugin
+} from "@alchemy/aa-accounts";
+import { AlchemyProvider } from "@alchemy/aa-alchemy";
+import { Address, UserOperationOverrides } from "@alchemy/aa-core";
+import { useCallback, useMemo } from "react";
+
+export const usePlugin = <AD, PD>(
+  provider: AlchemyProvider,
+  plugin: Plugin<AD, PD>,
+  pluginAddress_?: Address,
+) => {
+  if (!provider.isConnected<IMSCA>()) {
+    throw new Error(
+      "Provider must be connected to an MSCA before using this hook!"
+    );
+  }
+
+  const {
+    accountMethods,
+    providerMethods,
+    meta: { addresses },
+  } = plugin;
+
+  const pluginAddress = useMemo(() => pluginAddress_ ?? addresses[provider.rpcClient.chain.id], [pluginAddress_, addresses, provider]);
+
+  const { uninstallPlugin: uninstallPlugin_ } =
+    useMemo(() => pluginManagerDecorator(provider), [provider]);
+
+  const uninstallPlugin = useCallback(
```

Comment:
for consistency, would be better to also have `installPlugin` return by the hook which calls `installSessionKeyPlugin` internally

Reply:
that's tough to do in such a generic hook. I think instead we should add `uninstallSessionKeyPlugin` via the plugingen and then remove the uninstall method from here

Reply:
that works too